File: frontend/types/notification.ts
export type NotificationEventType = 'parent_linked' | 'message_received' | 'result_published' | 'student_absent' | 'assignment_created' | 'fee_reminder' | 'announcement_published';
export interface AppNotification { id: string; event_type: NotificationEventType; title: string; body: string; data: Record<string, unknown>; is_read: boolean; read_at?: string | null; created_at: string }
export interface NotificationPage { items: AppNotification[]; total: number; page: number; per_page: number; unread_count: number }

File: backend/app/schemas/notification.py
import uuid
from datetime import datetime
from pydantic import BaseModel
from app.models.notification import DomainEventType

class NotificationResponse(BaseModel):
    id: uuid.UUID; event_type: DomainEventType; title: str; body: str; data: dict; is_read: bool; read_at: datetime | None; created_at: datetime
    model_config = {"from_attributes": True}
class NotificationPage(BaseModel):
    items: list[NotificationResponse]; total: int; page: int; per_page: int; unread_count: int
class NotificationUnreadCount(BaseModel): count: int
